pkg/retriever/proposal.go

```diff
@@ -1,21 +1,38 @@
 package retriever
 
 import (
-	"github.com/filecoin-project/go-fil-markets/retrievalmarket"
-	"github.com/filecoin-project/go-fil-markets/shared"
+	"sync/atomic"
+	"time"
+
+	retrievaltypes "github.com/filecoin-project/go-retrieval-types"
 	"github.com/ipfs/go-cid"
 	"github.com/ipld/go-ipld-prime"
 	selectorparse "github.com/ipld/go-ipld-prime/traversal/selector/parse"
 )
 
-var dealIdGen = shared.NewTimeCounter()
+var dealIdGen = NewTimeCounter()
```

Comment:
sad this doesn't get shared, it's going to either get reimplemented in many places, or done badly like it was in autoretrieve

Reply:
can resolve in future pr